Realm/Tests/QueryTests.m

```diff
@@ -2155,7 +2155,6 @@ - (void)testDictionaryMultiLevelLinkQuery {
     XCTAssertThrows([CircleDictionaryObject objectsInRealm:realm where:@"ANY data = '2'"]);
     XCTAssertThrows([CircleDictionaryObject objectsInRealm:realm where:@"ANY circles.next = '2'"]);
     XCTAssertThrows([CircleDictionaryObject objectsInRealm:realm where:@"ANY data.circles = '2'"]);
-    XCTAssertThrows([CircleDictionaryObject objectsInRealm:realm where:@"circles.data = '2'"]);
```

Comment:
What does this query now do instead of throwing an exception? There should be a new test replacing this one.